dll/win32/ntdll_vista/threadpool.c

```diff
@@ -0,0 +1,2128 @@
+/*
+ * Thread pooling
+ *
+ * Copyright (c) 2006 Robert Shearman
+ * Copyright (c) 2014-2016 Sebastian Lackner
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+/*
+ * Old thread pooling API
+ */
+
+ /* INCLUDES *****************************************************************/
+
+#include <rtl_vista.h>
```

Comment:
Looking at the diff to Wine for this file, I see the following changes:
* Use `rtl_vista.h` and `debug.h` instead of Wine headers
* Typing changes (int vs long, RTL_CRITICAL_SECTION vs CRITICAL_SECTION etc)
* `GetCurrentProcess()` -> `(HANDLE)NtCurrentProcess()`
* `GetProcessHeap()` -> `NtCurrentPeb()->ProcessHeap`
* `TRACE` -> `DPRINT`

I'm assuming the GetCurrentProcess/GetProcessHeap changes are to avoid kernel32 dependencies? If so, that makes sense, though perhaps a `#define` would make future syncs easier.
I'm not sure about using DPRINTs instead of Wine tracing. Is this also due to dependencies?
Regarding the includes and types -- I would prefer if we could use Wine headers instead. This should automatically solve most of the typing issues (`LONG` will be `int` etc). Is that possible or do you encounter problems with that? Remember to define `__WINESRC__` for this file to get the right Wine-internal behavior.

Reply:
> I'm assuming the GetCurrentProcess/GetProcessHeap changes are to avoid kernel32 dependencies?
Yes, kinda circular reference appearing. I also thought about #define but it looks creapy

>I'm not sure about using DPRINTs instead of Wine tracing....
i trying to add same headers as in other files where Wine tracing used but stuck and just replaced to DPRINT1 :) Ok, i trying to make Wine tracing work here

Reply:
Yeah that seems to be a bug in our headers. Feel free to fix it to use the `typedef` if that helps.

Reply:
Things not so easy. I start making this CS fix but there are a lot of places where used Spare[0] from our current `CRITICAL_SECTION_DEBUG`. **union** way is not suitable because of initializations with `{ (DWORD_PTR)(__FILE__ ": somestring") } ` - with union this piece have to be wrapped in one more {} braces

This is a source of troubles :)
`//#ifdef __WINESRC__ //not all wine code is marked so
	DWORD_PTR Spare[8/sizeof(DWORD_PTR)];/ *  in Wine they store a string here * /
//#else
	//WORD SpareWORD;
//#endif`

I have one idea (#defining __HYBRIDSRC__ in dll header and in struct) for dlls with code from Wine and ours but sure this will blew up a discussion and i propose to do it in new PR

Reply:
You can define `__WINESRC__` for just one file.
```
set_source_files_properties(file.c PROPERTIES COMPILE_DEFINITIONS __WINESRC__)
```

Reply:
Ok, but anyway i think CS-fix should be in separate PR

Reply:
i noticed that almost all modules with `__WINESRC__` in CMake-file is a straight Wine copy and have this `include_directories(${REACTOS_SOURCE_DIR}/sdk/include/reactos/wine)`